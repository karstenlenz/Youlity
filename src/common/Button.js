import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

Button.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.string.isRequired,
}

export default function Button({ children, onClick }) {
  return <BlockButton onClick={onClick}>{children}</BlockButton>
}

const BlockButton = styled.button`
  display: block;
  padding: 10px;
  width: 100%;
  font-size: 18px;
`
