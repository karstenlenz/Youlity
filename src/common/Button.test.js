import React from 'react'
import { render, screen } from '@testing-library/react'
import Button from './Button'

describe('The Button component', () => {
  it('renders correctly', () => {
    const { container } = render(<Button>Test-Text</Button>)
    expect(container).toMatchSnapshot()
  })

  it('display the text "Test-Text"', () => {
    render(<Button>Test-Text</Button>)
    expect(screen.getByText('Test-Text')).toBeInTheDocument()
  })
})
