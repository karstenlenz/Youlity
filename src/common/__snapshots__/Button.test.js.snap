// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button component renders correctly 1`] = `
<div>
  <button
    class="Button__BtnStyled-sc-129yfz5-0 dlyhBD"
    width="100"
  >
    Test-Text
  </button>
</div>
`;
